src/Entries/Collection.php

```diff
@@ -602,6 +603,12 @@ public function delete()
     {
         $this->queryEntries()->get()->each->delete();
 
+        if ($this->structure()) {
+            $this->structure()->trees()->each(function ($tree) {
+                app(CollectionTreeRepository::class)->delete($tree);
```

Comment:
Can't you just call `$tree->delete()` there? It'd also dispatch the deleted event.

Reply:
Thanks Jelle, sounds like an improvement. I’ll look into it soon.

Reply:
Yeah you should be able to to `$this->structure()->trees()->each->delete();`